File: src/types/sqlDraft.ts
export interface SqlDraft {
  id: string
  connectionId?: string | null
  connectionNameSnapshot?: string | null
  database?: string | null
  schema?: string | null
  title: string
  sql: string
  createdAt: string
  updatedAt: string
  lastOpenedAt?: string | null
  closedAt?: string | null
}

export interface UpsertSqlDraftInput {
  id?: string | null
  connectionId?: string | null
  connectionNameSnapshot?: string | null
  database?: string | null
  schema?: string | null
  title: string
  sql: string
  closed?: boolean | null
}

File: src-tauri/src/models/sql_draft.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SqlDraft {
    pub id: Uuid,
    pub connection_id: Option<Uuid>,
    pub connection_name_snapshot: Option<String>,
    pub database: Option<String>,
    pub schema: Option<String>,
    pub title: String,
    pub sql: String,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
    pub last_opened_at: Option<DateTime<Utc>>,
    pub closed_at: Option<DateTime<Utc>>,
}
